### DISCUSSION TITLE
How to randomize items on page load and keep order?

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#18802] [Questions]
Let’s say we have fetched data from an API and now on our nuxt-app we want to display it like `<li v-for="item in randomizedItems" :key="item.key">…</li>`. `randomizedItems` is a computed property that randomizes the order of the fetched data. Everything works good so far, but as soon as we leave the page and then return to it, the order of the items will be changed (randomized) again.

How can we force randomization only once on page load?

Do I need to use `useState` or pinia?

**Comment by TheAlexLichter** [comment]
Exactly. The best way would be to "save" a seed in the state (e.g. via `useState`) on page load and reuse that, so e.g. a value generated by `Math.random` that will then be used in the function that sorts/shuffles.

**Reply by ghost** [reply]
@manniL Because I only want it happen client side (using SSG) I’ve implemented the following code:

```js
const reorderedItems = process.client
    ? useState('reorderedItems', () => {
        // some more complex shuffling happening here, but for this example we keep it simple
        return shuffle(items.value);
    })
    : items;
```

Is there a better practice? Does it affect performance?

**Reply by TheAlexLichter** [reply]
Oh, if you only want to have it on client-side, things will be a bit difficult as you might see hydration errors. Contents *have* to be the same on server and client, unless you only render them on the client via `<ClientOnly>`.

**Reply by ghost** [reply]
@manniL I don’t have any errors with the pattern above

**Reply by TheAlexLichter** [reply]
No hydration errors or anything? Interesting 🤔

**Reply by ghost** [reply]
@manniL nope. i’ve tested it again … no hydration errors, no errors … works fine. The only thing is that it flickers when entering the page, because it is showing the server-rendered version for a millisecond, before the shuffled client side version renders … Is there any "best practice" for this use case: randomize list client side on page load using SSG?
